src/Workspaces/Core/Portable/FlowAnalysis/LValueFlowCaptureProvider.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.CodeAnalysis.FlowAnalysis
+{
+    /// <summary>
+    /// Helper class to detect <see cref="IFlowCaptureOperation"/>s that are l-value captures.
+    /// </summary>
+    internal static class LValueFlowCapturesProvider
+    {
+        private static readonly ConditionalWeakTable<ControlFlowGraph, ImmutableHashSet<CaptureId>> s_lValueFlowCapturesCache =
+            new ConditionalWeakTable<ControlFlowGraph, ImmutableHashSet<CaptureId>>();
```

Comment:
this seems terrifying. Who needs this functionality, why can they not just cache data accordingly instead of needing this under the covers cache?

Reply:
I was being defensive about multiple analyzers asking for this same information about same CFG. I will remove it for now, and we can add it if we indeed see perf problems here.

---
In reply to: [230606769](https://github.com/dotnet/roslyn/pull/30777#discussion_r230606769) [](ancestors = 230606769)